test/ClientTest.cpp

```diff
@@ -557,6 +558,82 @@ single_conn_call(Connector<BUFFER, NetProvider> &client)
 	client.close(conn);
 }
 
+/* Authentication */
+template <class BUFFER, class NetProvider = Net_t>
+void
+authentication_test(Connector<BUFFER, NetProvider> &client)
+{
+	TEST_INIT(0);
+	TEST_CASE("Authentication within user_name and password");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == 0);
+		client.close(conn);
+	}
+	
+	TEST_CASE("Authentication with user_name and password good");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastas", "123456"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == 0);
+		client.close(conn);
+	}
+	
+	TEST_CASE("Authentication with user_name and wrong password");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastas", "1234567"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == -1);
+		client.close(conn);
+	}
+
+
+	TEST_CASE("Authentication with wrong user_name");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastasius", "123456"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == -1);
+		client.close(conn);
+	}
+
+	TEST_CASE("Authentication with some privileges without errors");
+	{
+		Connection<Buf_t, NetProvider> conn(client);
+		Config config = {localhost, port, "Anastas", "123456"};
+		int rc = client.connect(conn, config);
+		fail_unless(rc == 0);
+		uint32_t space_id = 512;
+		std::tuple data = std::make_tuple(666, "111", 1.01);
+		rid_t f1 = conn.space[space_id].replace(data);
+		data = std::make_tuple(777, "asd", 2.02);
+		rid_t f2 = conn.space[space_id].replace(data);
```

Comment:
Authentication requires a bit more test cases. You can look at examples from Tarantool repo